### PR TITLE
Dispatch and more compute states

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

**Comment by kvark** [comment]
bors r=glrovesNL

**Comment by kvark** [comment]
Published in https://crates.io/crates/metal-rs/0.6.4
Travis OSX is not in the mood lately...

File: src/encoder.rs [hunk 1]
```diff
@@ -434,9 +438,41 @@ impl ComputeCommandEncoderRef {
         }
     }
 
+    pub fn set_texture(&self, index: u64, texture: Option<&TextureRef>) {
```
Comment (JohnColanduoni) [inline src/encoder.rs]: The u64 should be an NSUInteger.

File: src/encoder.rs [hunk 2]
```diff
@@ -434,9 +438,41 @@ impl ComputeCommandEncoderRef {
...
+        unsafe {
+            msg_send![self, setTexture:texture
+                               atIndex:index]
+        }
+    }
+
+    pub fn set_sampler_state(&self, index: u64, sampler: Option<&SamplerStateRef>) {
```
Comment (JohnColanduoni) [inline src/encoder.rs]: Same here

**Comment by kvark** [comment]
Thanks @JohnColanduoni ! The notes are addressed in #28

**Pull request merged**